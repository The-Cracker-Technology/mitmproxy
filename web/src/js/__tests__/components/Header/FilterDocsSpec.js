import React from 'react'
import renderer from 'react-test-renderer'
import FilterDocs from '../../../components/Header/FilterDocs'

describe('FilterDocs Component', () => {
    let mockResponse = { json:
            () => { return { commands: [['cmd1', 'foo'], ['cmd2', 'bar']]}}
        },
        promise = Promise.resolve(mockResponse)
    global.fetch = r => { return promise }

    let filterDocs = renderer.create(<FilterDocs/>),
        tree = filterDocs.toJSON()

    it('should render correctly when fetch success', () => {
        expect(tree).toMatchSnapshot()
    })
})
