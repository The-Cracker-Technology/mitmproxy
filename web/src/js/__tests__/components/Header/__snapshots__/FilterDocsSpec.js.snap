// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDocs Component should render correctly when fetch success 1`] = `
<i
  className="fa fa-spinner fa-spin"
/>
`;
